### ISSUE TITLE
NextJs dynamic path variables do not appear in router.query on initial page load, when basePath is changed and middleware exists

### ISSUE THREAD

**Issue by tasfik007** [vercel/next.js#52051]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: linux
      Arch: x64
      Version: #23-Ubuntu SMP PREEMPT_DYNAMIC Wed May 17 16:55:20 UTC 2023
    Binaries:
      Node: 16.13.0
      npm: 8.1.0
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 12.3.0
      eslint-config-next: 11.1.0
      react: 17.0.2
      react-dom: 17.0.2

warn  - Latest canary version not detected, detected: "12.3.0", newest: "13.4.8-canary.13".
        Please try the latest canary version (`npm install next@canary`) to confirm the issue still exists before creating a new issue.
        Read more - https://nextjs.org/docs/messages/opening-an-issue
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

Middleware / Edge (API routes, runtime), Routing (next/router, next/navigation, next/link)

### Link to the code that reproduces this issue or a replay of the bug

https://github.com/tasfik007/subpath-test

### To Reproduce

1. Initialize a NextJS application
2. Change the basePath in the .next.config.js file
3. Add a middleware.js file

### Describe the Bug

I have a NextJS application, recently I changed the basePath of my application in next.config.js file just added the basePath key with my desired path.

After doing that I noticed dynamic path variables that were passed through slugs (e.g. /something/[id].js) are not getting loaded initially in next's router.query (in this case id), but after a full page reload it appears perfectly. Surprisingly the variables were also accessible in the getServerSideProp's context.

Now this only happens if I have a middleware file (middleware.js) in my project. This issue happens even if I do nothing in the middleware.

### Expected Behavior

Dynamic variables that are passed through the slugs should be loaded in router.query first time user loaded on the page.

### Which browser are you using? (if relevant)

Firefox Browser - 114.0.2 (64-bit)

**Comment by BrunoFerreira95** [comment]
hmmm, one question in the url after run the project the path dynamic, show, in the browser, when you acesses the route ?

**Comment by tasfik007** [comment]
> hmmm, one question in the url after run the project the path dynamic, show, in the browser, when you acesses the route ?

yes, I can see the dynamic path variable in the route itself in the browser when I first land in.
